### PR TITLE
[lua] update to 5.4.4

### LINKED ISSUE #22890: [lua] update to 5.4.4
Library name: lua

New version number: 5.4.4

Other information that may be useful (release notes, etc...)

https://www.lua.org/news.html

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #22890 
[lua] update to 3.37.1
All features have been tested successfullly in the following triplet:

- x64-windows
- x64-windows-static
- x86-windows

### CODE REVIEW

File: ports/lua/portfile.cmake [hunk 1]
```diff
@@ -55,15 +54,15 @@ endif()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     if(VCPKG_TARGET_IS_WINDOWS)
-        file(READ ${CURRENT_PACKAGES_DIR}/include/luaconf.h LUA_CONF_H)
+        file(READ "${CURRENT_PACKAGES_DIR}/include/luaconf.h" LUA_CONF_H)
```
Comment (JonLiu1993) [inline ports/lua/portfile.cmake:57]: vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/luaconf.h" "defined(LUA_BUILD_AS_DLL)" "1")

File: ports/lua/portfile.cmake [hunk 2]
```diff
@@ -31,22 +31,21 @@ if(VCPKG_TARGET_IS_IOS AND "tools" IN_LIST FEATURES)
     message(FATAL_ERROR "lua[tools] is not supported for iOS platform build")
 endif()
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/lua/portfile.cmake:35]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/lua/portfile.cmake [hunk 3]
```diff
@@ -31,22 +31,21 @@ if(VCPKG_TARGET_IS_IOS AND "tools" IN_LIST FEATURES)
...
+vcpkg_cmake_install()
 
 vcpkg_copy_pdbs()
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-lua TARGET_PATH share/unofficial-lua)
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-lua CONFIG_PATH share/unofficial-lua)
 
 if("cpp" IN_LIST FEATURES)
-    vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-lua-cpp TARGET_PATH share/unofficial-lua-cpp)
+    vcpkg_cmake_config_fixup(CONFIG_PATH "share/unofficial-lua-cpp")
```
Comment (JonLiu1993) [inline ports/lua/portfile.cmake:48]:
```suggestion
    vcpkg_cmake_config_fixup(PACKAGE_NAME share/unofficial-lua-cpp CONFIG_PATH "share/unofficial-lua-cpp")
```

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the update!

**Pull request merged**